### PR TITLE
Help github review process for checkstyle

### PR DESCRIPTION
**Pull request by miurahr**
This github actions just run gradle checkstyleMain and checkstyleTest and send annotation to github API.

When reviewing Pull-Request on github project page, you can see checkstyle warnings on code diff page.

Signed-off-by: Hiroshi Miura <miurahr@linux.com>

### CODE REVIEW

**Comment by miurahr** [comment]
You can see how it looks on https://github.com/miurahr/omegat/pull/1/files

**Comment by amake** [comment]
Right now our CI is all Azure. I don't really want to add another platform into the mix.

I would prefer to either move all the Azure stuff here, or improve the Azure stuff.

**Comment by miurahr** [comment]
Close here because I don't know Azure configurations.

**Comment by miurahr** [comment]
Now we use Github as a truth of code.

File: .github/workflows/checkstyle-annotate.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Run checkstyle
+
+on: pull_request
+
+jobs:
+  checkstyle:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (miurahr) [inline .github/workflows/checkstyle-annotate.yml:9]:
```suggestion
    - uses: actions/checkout@v3
```

File: .github/workflows/checkstyle-annotate.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    - uses: actions/setup-java@v2.4.0
+      with:
+        distribution: 'temurin'
+        java-version: '8'
```
Comment (miurahr) [inline .github/workflows/checkstyle-annotate.yml:13]:
```suggestion
        java-version: '11'
```

**Review by miurahr — COMMENTED** [review]
Need to update versions.

File: .github/workflows/checkstyle-annotate.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name: Run checkstyle
+
+on: pull_request
+
+jobs:
+  checkstyle:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v2.4.0
```
Comment (miurahr) [inline .github/workflows/checkstyle-annotate.yml:10]:
```suggestion
    - uses: actions/setup-java@v3
```

File: .github/workflows/checkstyle-annotate.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        java-version: '11'
+    - uses: kiancross/checkstyle-annotations-action@v1
+    - name: run checkstyle
+      uses: gradle/gradle-build-action@v2.1.2
```
Comment (miurahr) [inline .github/workflows/checkstyle-annotate.yml:16]:
```suggestion
      uses: gradle/gradle-build-action@v2
```

It is better to accept improvements of minor version up.

**Pull request merged**